### ISSUE TITLE
Support the conditional configuration mode of the native-image-agent

### ISSUE THREAD

**Issue by gradinac** [graalvm/native-build-tools#206]
The goal of this ticket is to serve as a discussion point for supporting the conditional configuration generation mode of the native-image-agent.

Once https://github.com/oracle/graal/pull/4270 is merged, the native-image agent will support automatically generating conditional native-image configuration. This mode will mostly be useful for library authors, as well as creators of component based applications (i.e. you can exclude an optional component from the app and automatically have all configuration from that component be excluded as well).

The intended way to generate conditional configuration automatically is through unit tests: users would run all of their tests with the agent attached and configured in this mode and as a result get a best-effort conditional configuration that can then be tweaked manually.

There are two different ways we could support this:
 - Adding a new way of running tests (e.g. new task) that would execute all of the user's current tests with JVM options that enable the agent.
 - As discussed with @melix, we could rework our current agent support and introduce the concept of agent modes: https://github.com/graalvm/native-build-tools/pull/190#issuecomment-1030592573.
 
Currently, there are only two different parameters the users can configure when using the agent in this mode:
 - The packages on which the conditions will be generated - if the user specifies `org.acme`, all conditions will be based on types from `org.acme`
 - A regex list that will be used to filter out classes from the configuration - any class in the configuration or condition that matches any of the specified regex expressions will be removed from the config.
Depending on user feedback, we may introduce additional parameters that will tune the config generation process.

We could possibly introduce default class filters (e.g. filter out all `MockitoMock` classes or classes that contain `Test` in the name) and a way to turn these off. These could either be always active or only included if certain dependencies are used.

**Comment by gradinac** [comment]
The best way to support this feature would be to introduce agent modes as described in the issue. The modes would not map directly to an agent command line, but could run arbitrarily complex steps in the background. The proposed format, based on
https://github.com/graalvm/native-build-tools/pull/190#issuecomment-1030592573, would be:
```
<agent>
    <mode name="modeName">
        <modeSpecificOption></modeSpecificOption>
    </mode>
    <options>
        <option>some common option</option>
    </options>
</agent>
``` 
The agent configuration block would be divided into two parts:
 - Mode block
 - Common options block

The **mode block** would select one of the available modes, as well as provide a way to pass additional arguments specific to that mode.
Initially, we would have the following modes available:
 - `standard`: the agent collects configuration, automatically merging it with `native-image-configure`  in case of multiple JVM executions
 - `raw`: adds the agent to the JVM execution but does not do any additional logic. Allows users to pass arbitrary flags to the agent
 - `trace`: runs the agent in the tracing mode and provides the user with all created trace files (one per JVM execution)
 - `origins`: runs the agent and provides configuration with origin information. This mode is only useful for debugging
 - `conditional`: runs the agent and collects conditional configuration. The agent would collect metadata for each JVM run that will then be used by `native-image-configure` to generate the conditional configuration.

The **common options block** would allow users to specify options that are usable with every agent mode, such as caller/access filter files.
